cabot/cabotapp/tests/test_status_check.py

```diff
@@ -383,3 +386,53 @@ def test_activity_counter_reset_and_save(self, mock_now, mock_save):
         self.assertEqual(self.counter.last_disabled, now)
         self.assertEqual(mock_now.call_count, 1)
         self.assertEqual(mock_save.call_count, 1)
+
+
+class TestRunWindow(LocalTestCase):
+    def test_empty_run_window(self):
+        self.http_check.run_window = CheckRunWindow([])
+        self.assertTrue(self.http_check.should_run())
+
+    @patch('cabot.cabotapp.models.timezone.now')
+    def test_run_window(self, mock_now):
+        start = time(12, 0, 0)
+        end = time(13, 30, 0)
+
+        weekdays = (rrule.MO, rrule.TU, rrule.WE, rrule.TH, rrule.FR, rrule.SA, rrule.SU)
+        recurrence = rrule.rrule(rrule.WEEKLY, interval=1, byweekday=weekdays)
+        self.http_check.run_window = CheckRunWindow([CheckRunWindow.Window(start_time=start, end_time=end,
+                                                                           recurrence=recurrence)])
+
+        active = [
+            datetime(2019, 10, 20, 12, 0, 0),
+            datetime(2019, 10, 20, 12, 5, 0),
+            datetime(2019, 10, 20, 13, 25, 0),
+            datetime(2019, 10, 20, 13, 30, 0),
+
+            # also test tomorrow
+            datetime(2019, 10, 21, 12, 0, 0),
+            datetime(2019, 10, 21, 13, 0, 0),
+            datetime(2019, 10, 21, 13, 30, 0),
+            datetime(2040, 10, 21, 12, 0, 0),  # extremely tomorrow
```

Comment:
haha

Time's arrow neither stands still nor reverses. It merely marches forward.